channel/ledger.go

```diff
@@ -0,0 +1,65 @@
+package channel
+
+import (
+	"errors"
+
+	"github.com/statechannels/go-nitro/channel/state"
+)
+
+const proposerIndex = uint(0)
+
+type TwoPartyLedger struct {
+	Channel
+}
+
+// NewTwoPartyLedger constructs a new two party ledger channel from the supplied state.
+func NewTwoPartyLedger(s state.State, myIndex uint) (*TwoPartyLedger, error) {
+	if myIndex > 1 {
+		return &TwoPartyLedger{}, errors.New("myIndex in a two party ledger channel must be 0 or 1")
+	}
+	if len(s.Participants) != 2 {
+		return &TwoPartyLedger{}, errors.New("two party ledger channels must have exactly two participants")
+	}
+
+	c, err := New(s, myIndex)
+
+	return &TwoPartyLedger{*c}, err
+}
+
+// Equal returns true if the supplied TwoPartyLedger is deeply equal to the receiver, false otherwise.
+func (lc *TwoPartyLedger) Equal(lc2 *TwoPartyLedger) bool {
+	return lc.Channel.Equal(lc2.Channel)
+}
+
+// Clone returns a pointer to a new, deep copy of the receiver, or a nil pointer if the receiver is nil.
+func (lc *TwoPartyLedger) Clone() *TwoPartyLedger {
+	if lc == nil {
+		return nil
+	}
+	w := TwoPartyLedger{*lc.Channel.Clone()}
+	return &w
+}
+
+// Proposed returns the latest proposed ledger state that is only signed by the proposer.
+// If the latest state signed by the proposer is the supported state then Proposed returns a nil state and false.
+func (lc *TwoPartyLedger) Proposed() (state.State, bool) {
+
+	highestSignedByProposer := uint64(0)
+
+	for turnNum, signedState := range lc.SignedStateForTurnNum {
+		if signedByProposer := signedState.HasSignatureForParticipant(proposerIndex); signedByProposer && turnNum > highestSignedByProposer {
+			highestSignedByProposer = turnNum
+		}
+	}
+
+	if highestSignedByProposer == lc.latestSupportedStateTurnNum {
+		return state.State{}, false
+	} else {
+		return lc.SignedStateForTurnNum[highestSignedByProposer].State(), true
+	}
+}
+
+// IsProposer returns true if the we are responsible for proposing ledger updates, false otherwise
+func (lc *TwoPartyLedger) IsProposer() bool {
```

Comment:
> Language here is a little confusing. It returns whether the current client is a proposer, not whether the ledger is a proposer.

I tried to follow a similar naming convention to [isAlice](https://github.com/statechannels/go-nitro/blob/ag/leader-follower-ledger/protocols/virtualfund/virtualfund.go#L591:L591). In this case we're returning if a participant in a ledger channel is the proposer or not.

Happy to rename this if it would be clearer!

Reply:
You are right that this follows an established convention.